credentials/apps/badges/models.py

```diff
@@ -83,6 +83,25 @@ def save(self, *args, **kwargs):
     @classmethod
     def by_uuid(cls, template_uuid):
         return cls.objects.filter(uuid=template_uuid, origin=cls.ORIGIN).first()
+    
+    def user_progress(self, username: str) -> float:
+        """
+        Calculate user progress for badge template.
+        """
+        requirements_count = BadgeRequirement.objects.filter(template=self).count()
```

Comment:
Do not check requirements here at all, we don't care if the badge template is not configured yet.